cmd/tailscale/cli/up.go

```diff
@@ -388,9 +388,11 @@ func updatePrefs(prefs, curPrefs *ipn.Prefs, env upCheckEnv) (simpleUp bool, jus
 	if !env.upArgs.reset {
 		applyImplicitPrefs(prefs, curPrefs, env)
 
-		if err := checkForAccidentalSettingReverts(prefs, curPrefs, env); err != nil {
+		su, err := checkForAccidentalSettingReverts(prefs, curPrefs, env)
+		if err != nil {
 			return false, nil, err
 		}
+		simpleUp = su
```

Comment:
Optional: I think we don't need the temporary `su` variable or to shadow `err`, and (subjectively) this is a bit easier to read:
```suggestion
		if simpleUp, err = checkForAccidentalSettingReverts(prefs, curPrefs, env); err != nil {
			return false, nil, err
		}
```
(or keep the assignment on its own line before the `if` statement, if you prefer)